source/dagger/core/game/finite_state_machine.h

```diff
@@ -52,7 +52,7 @@ struct FSM
 	}
 
 protected:
-	const UInt32 stateSize = (UInt32) std::numeric_limits<enum States>::max();
+	const UInt32 stateSize = (UInt32) std::numeric_limits<States>::max();
```

Comment:
In file included from ../source/dagger/gameplay/platformer/character_controller_fsm.h:3:
../source/dagger/core/game/finite_state_machine.h:55:61: error: declaration of 'States' shadows template parameter
        const UInt32 stateSize = (UInt32) std::numeric_limits<enum States>::max();
                                                                   ^
../source/dagger/core/game/finite_state_machine.h:8:19: note: template parameter is declared here
template<typename States, ENUM_ONLY(States)>
                  ^
../source/dagger/core/game/finite_state_machine.h:55:61: error: ISO C++ forbids forward references to 'enum' types
        const UInt32 stateSize = (UInt32) std::numeric_limits<enum States>::max();